apps/cowswap-frontend/src/modules/swap/services/swapFlow/index.ts

```diff
@@ -38,7 +40,10 @@ export async function swapFlow(
 
     const hooks = buildAppDataHooks([permitData])
 
-    input.orderParams.appData = addHooksToAppData(input.orderParams.appData, hooks)
+    input.orderParams.appData = updateHooksOnAppData(input.orderParams.appData, hooks)
+  } else {
+    // Otherwise, remove hooks (if any) from appData to avoid stale data
+    input.orderParams.appData = updateHooksOnAppData(input.orderParams.appData, undefined)
```

Comment:
it feels strange the logic of deleting the hook info is here. Is the responsability of the `appData` module, not of the `swap/services/swapFlow`

Reply:
That is being done by the appData module.
It takes the appData instance and updates/removes the hooks as needed.
It's stored in the order params.